### PR TITLE
Fixes #583: Move to Ansible-core for CI tests

### LINKED ISSUE #583: CI does not run sanity tests with ansible-core 2.11 and ansible-core's devel branch
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
It currently only runs these tests with ansible-base 2.10, which is not a good idea at all.

Check out the testing requirements for inclusion in Ansible: https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing

### PR DESCRIPTION
**Pull request by rodvand**
- Moved from ansible-base to ansible-core
- Removed form_factor from interface and family from ip_address as they were marked deprecated/removed
- Add test for pynetbox/requests library in nb_lookup.py
- Substituted list comprehension to match sanity test
- Removed family from test data to match removal from module
- Ignore pylint test on nb_lookup due to bug https://github.com/PyCQA/pylint/issues/3507 in pylint 2.6.0 which is used in ansible-test

### CODE REVIEW

**Comment by ryanmerolle** [comment]
Closes #583

**Review by ryanmerolle — APPROVED** [review]

**Review by sc68cal — DISMISSED** [review]

File: plugins/modules/netbox_device_interface.py [hunk 1]
```diff
@@ -298,9 +290,6 @@ def main():
                 options=dict(
                     device=dict(required=False, type="raw"),
                     name=dict(required=True, type="str"),
-                    form_factor=dict(
```
Comment (felixfontein) [inline plugins/modules/netbox_device_interface.py:301]: Removing options requires a new major release, which I would really avoid for now until you have a new minor (or patch) release that fixes the current issues. A new major release has to wait for Ansible 5 and cannot be included in the next Ansible 4 release (which will happen ~next Tuesday BTW).

File: plugins/modules/netbox_ip_address.py [hunk 2]
```diff
@@ -323,12 +312,6 @@ def main():
                 type="dict",
                 required=True,
                 options=dict(
-                    family=dict(
```
Comment (felixfontein) [inline plugins/modules/netbox_ip_address.py:326]: Same here.

**Review by felixfontein — COMMENTED** [review]
I'd suggest to remove the removals from this PR and wait with them for the next major release.

**Comment by felixfontein** [comment]
(Also you'd need a `removed_features` changelog fragment.)

File: plugins/modules/netbox_device_interface.py (hunk 1 above)
Comment (rodvand) [reply plugins/modules/netbox_device_interface.py:301]: Thanks @felixfontein , this is great feedback! I did try to add the `'removed_from_collection': 'netbox.netbox'` but then the tests said `ERROR: plugins/modules/netbox_device_interface.py:0:0: collection-deprecated-version: Argument 'form_factor' in argument_spec found in data has a deprecated removed_in_version '0.3.0', i.e. the version is less than or equal to the current version of this collection (3.1.1)`.

Does this mean it should really have been removed before? And should I now just bump the `removed_in_version` to 4.0.0 to resolve this as smooth as possible? And then I guess the tests will prompt me if I'm not removing it from version 4.0.0?

Comment (felixfontein) [reply plugins/modules/netbox_device_interface.py:301]: Yes, it should have been removed in 1.0.0. (Or maybe the value '0.3.0' is also wrong, maybe it was the version when the deprecation was added?)

In any case, you could either add an ignore.txt entry for this, or replace 0.3.0 by 4.0.0, as you suggested. In the latter case, you (or whoever else) will be reminded once you change the version to 4.0.0 and run sanity checks.

Comment (ryanmerolle) [reply plugins/modules/netbox_device_interface.py:301]: This is super useful feedback!  Thanks @felixfontein !

File: plugins/modules/netbox_ip_address.py (hunk 2 above)
Comment (ryanmerolle) [reply plugins/modules/netbox_ip_address.py:326]: Again Thanks!

**Comment by ryanmerolle** [comment]
@rodvand, do you think you would be able to address @felixfontein comments.  I will work with @FragmentedPacket and @jeremystretch to try to get a release cut and pushed to ansible galaxy before the next release of Ansible 4 this up coming Tuesday.

**Comment by rodvand** [comment]
> @rodvand, do you think you would be able to address @felixfontein comments. I will work with @FragmentedPacket and @jeremystretch to try to get a release cut and pushed to ansible galaxy before the next release of Ansible 4 this up coming Tuesday.

Working on it.

**Review by ryanmerolle — APPROVED** [review]
Looks good to me.

@felixfontein @gundalow @rodvand any final comments or items before I merge?

**Review by felixfontein — APPROVED** [review]
LGTM (as far as I can judge :) )

**Pull request merged**